cmd/yggdrasil/main.go

```diff
@@ -53,12 +52,11 @@ func main() {
 	getpkey := flag.Bool("publickey", false, "use in combination with either -useconf or -useconffile, outputs your public key")
 	loglevel := flag.String("loglevel", "info", "loglevel to enable")
 	flag.Parse()
-
-	// Catch interrupts from the operating system to exit gracefully.
-	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
-
-	// Capture the service being stopped on Windows.
-	minwinsvc.SetOnExit(cancel)
+	
+	done := make(chan struct{})
+	defer close(done)
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
Would probably prefer to keep the `signal.NotifyContext` approach if it achieves the same result, as we're likely to use contexts more in the future.

Reply:
Done.